libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/ui/SalesforceReactActivity.java

```diff
@@ -204,6 +204,7 @@ public void authenticatedRestClient(RestClient client) {
     public void logout(Callback successCallback) {
         SalesforceReactLogger.i(TAG, "logout called");
         SalesforceSDKManager.getInstance().logout(this);
+        ReactBridgeHelper.invokeSuccess(successCallback, "logout success");
     }
```

Comment:
By being able to invoke a callback on logout from android, we can reset the logoutInitiated flag in the React Native context once the user is logged out without closing and reloading the app. See this PR for the react native changed https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/pull/73